packages/block-editor/src/components/iframe/index.js

```diff
@@ -280,6 +287,13 @@ function Iframe( {
 				colors the html element */
 				background-color: white;
 			}
+			/* 
+				Block placeholders are "technically" part of the editor UI,
+				and should therefore match the editor UI's directionality.
+			*/
+			.block-editor-block-list__layout .components-placeholder { 
```

Comment:
This change is optional (follow up?), and there's probably a better way.

The CSS rule here is in lieu of setting the `dir` inside the individual blocks.

With a site lang of `ar` and a user lang of `en`:

 | Before | After |
|--------|--------|
| <img width="396" alt="Screenshot 2024-07-25 at 11 27 36 AM" src="https://github.com/user-attachments/assets/c8a87f04-6d41-4933-9f13-a774433a2041"> | <img width="378" alt="Screenshot 2024-07-25 at 11 27 45 AM" src="https://github.com/user-attachments/assets/61a9feab-2773-4e69-91d6-d09f5519ed5f"> | 

With a site lang of `en` and a user lang of `ar:

 | Before | After |
|--------|--------|
|  <img width="369" alt="Screenshot 2024-07-25 at 11 42 05 AM" src="https://github.com/user-attachments/assets/47f280f5-83a6-453b-a335-938182afa793"> | <img width="380" alt="Screenshot 2024-07-25 at 11 41 36 AM" src="https://github.com/user-attachments/assets/517a58c8-f352-43f7-a744-a305fdc652d2"> |

Reply:
Maybe we could compose `<Placeholder />` for blocks and do all the locale trickery in the higher order component?

E.g., `const BlockPlaceholder = withLocale( Placeholder )`, or whatever it is